core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -95,8 +102,8 @@ public UnderStoreBlockInStream(long initPos, long length, long fileBlockSize,
 
   @Override
   public void close() throws IOException {
-    mUnderStoreStreamFactory.close();
     mUnderStoreStream.close();
+    mUnderStoreStreamFactory.close();
```

Comment:
does the order matter? if yes, please comment here why we want to make the order like this (so we don't mess it up in the future)